compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/Constants.java

```diff
@@ -40,4 +40,5 @@ private Constants() {
 
     public static final int INIT_METHOD_SPLIT_SIZE = 50;
 
+    public static final String UNICODE_REGEX = "\\\\u[{]([a-fA-F0-9]*)[}]";
```

Comment:
do we need `[{]` and `[}]` instead we should be able to use `\\{` and `\\}`

Reply:
And please check what happens if an empty string is matched here and too large number matches here.
I think it will be a number format exception when parsing the int.
Shall we therefore use something like `\\\\u\\{([a-fA-F0-9]{1,6})\\}`

Reply:
> do we need `[{]` and `[}]` instead we should be able to use `\\{` and `\\}`

+1

Reply:
> And please check what happens if an empty string is matched here and too large number matches here.
> I think it will be a number format exception when parsing the int.
> Shall we therefore use something like `\\\\u\\{([a-fA-F0-9]{1,6})\\}`

According to the spec, there should be at least one hex value. So if the user enters an empty string, parse throws an error. Also, the Unicodes which are out of range(greater than 0x10FFFF) are handled [here](https://github.com/ballerina-platform/ballerina-lang/pull/20714/files#diff-87627a1d9b0a3201b74c0067bc4bdcbbR2661), because we need to throw a compile error.